chrome-extension/background.js

```diff
@@ -68,3 +68,21 @@ chrome.runtime.onInstalled.addListener(async () => {
 
   await chrome.storage.sync.set({ settings: defaultSettingCodes });
 });
+
+// background script
+chrome.runtime.onMessage.addListener(function (message, sender, senderResponse) {
```

Comment:
this is a valid way of dodging CORS issues, but two things:

- is this necessary? the headers you added should make it possible to just fetch from the client script
- if it is, can we merge with the above runtime listener, using the same format? Also try to use a more descriptive type name, rather than just "json"

Reply:
- Yes, it was necessary. I tried it with the headers and it didn't seem to fix anything. This is a common problem apparently even with the headers according to: https://medium.com/geekculture/bypassing-cors-with-a-google-chrome-extension-7f95fd953612. The workaround tutorial I followed is this: https://javascript.plainenglish.io/fetch-data-in-chrome-extension-v3-2b73719ffc0e
- Sure, I will merge over and fix that type name

Reply:
If corsproxy worked, then it shouldn't be necessary. Try with the preview deployment url. I'm pretty certain the medium article you linked is incorrect because they add the headers on the request side instead of the response, which is what we did by adding it to the next.config.js. 

I'm still at work so I can't verify if the headers send properly.

Reply:
I tried both, and neither worked. At least it doesn't work on my system. Maybe you could try getting it to work on your end? For now I think this workaround works pretty well and also looks a little more streamlined for future implementations.